pcsx2/PINE.cpp

```diff
@@ -406,108 +399,99 @@ PINEServer::IPCBuffer PINEServer::ParseCommand(char* buf, char* ret_buffer, u32
 			}
 			case MsgSaveState:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size))
 					goto error;
-				StateCopy_SaveToSlot(FromArray<u8>(&buf[buf_cnt], 0));
+				VMManager::SaveStateToSlot(FromArray<u8>(&buf[buf_cnt], 0));
 				buf_cnt += 1;
 				break;
 			}
 			case MsgLoadState:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size))
 					goto error;
-				StateCopy_LoadFromSlot(FromArray<u8>(&buf[buf_cnt], 0), false);
+				VMManager::LoadStateFromSlot(FromArray<u8>(&buf[buf_cnt], 0));
 				buf_cnt += 1;
 				break;
 			}
 			case MsgTitle:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
-				char* title = new char[GameInfo::gameName.size() + 1];
-				sprintf(title, "%s", GameInfo::gameName.ToUTF8().data());
-				const u32 size = strlen(title) + 1;
-				title[size] = 0x00;
+				const std::string gameName = VMManager::GetGameName();
+				const u32 size = gameName.size() + 1;
 				if (!SafetyChecks(buf_cnt, 0, ret_cnt, size + 4, buf_size))
 					goto error;
 				ToArray(ret_buffer, size, ret_cnt);
 				ret_cnt += 4;
-				memcpy(&ret_buffer[ret_cnt], title, size);
+				memcpy(&ret_buffer[ret_cnt], gameName.c_str(), size);
 				ret_cnt += size;
-				delete[] title;
 				break;
 			}
 			case MsgID:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
-				char* title = new char[GameInfo::gameSerial.size() + 1];
-				sprintf(title, "%s", GameInfo::gameSerial.ToUTF8().data());
-				const u32 size = strlen(title) + 1;
-				title[size] = 0x00;
+				const std::string gameSerial = VMManager::GetGameSerial();
+				const u32 size = gameSerial.size() + 1;
 				if (!SafetyChecks(buf_cnt, 0, ret_cnt, size + 4, buf_size))
 					goto error;
 				ToArray(ret_buffer, size, ret_cnt);
 				ret_cnt += 4;
-				memcpy(&ret_buffer[ret_cnt], title, size);
+				memcpy(&ret_buffer[ret_cnt], gameSerial.c_str(), size);
 				ret_cnt += size;
-				delete[] title;
 				break;
 			}
 			case MsgUUID:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
-				char* title = new char[GameInfo::gameCRC.size() + 1];
-				sprintf(title, "%s", GameInfo::gameCRC.ToUTF8().data());
-				const u32 size = strlen(title) + 1;
-				title[size] = 0x00;
+				const u32 size = 9;
+				char* crc = new char[size];
+				snprintf(crc, size, "%08x", VMManager::GetGameCRC());
```

Comment:
`const std::string crc(fmt::format("{:08X}", VMManager::GetGameCRC()));`

Then can be written back like the other strings.

Reply:
Fixed, and also think I replaced the `char[]` usage in `Version` with fmt correctly too.